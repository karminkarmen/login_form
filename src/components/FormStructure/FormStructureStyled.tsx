import styled from 'styled-components';
import { centered } from '../../configs/mixins';

const FormStructureStyled = styled.div`
    flex-direction: column;
    ${centered}
    border-radius:  ${props => props.theme.radius.maximum};
    width: ${props => props.theme.width.mainElementWidth};
    height: ${props => props.theme.width.mainElementWidth};
`;

export default FormStructureStyled;