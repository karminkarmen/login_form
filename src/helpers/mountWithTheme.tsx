import React from 'react';
import { mount } from 'enzyme';
import { ThemeProvider } from 'styled-components';
import theme from '../configs/theme';

const mountWithTheme = (child: React.ReactElement) => {
    return mount(child, {
        wrappingComponent: ({ children }) => <ThemeProvider theme={theme}>{children}</ThemeProvider>,
    });
}

export default mountWithTheme;