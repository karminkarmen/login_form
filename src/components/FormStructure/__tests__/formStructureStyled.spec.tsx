import React from 'react';
import FormStructureStyled from '../FormStructureStyled';
import mountWithTheme from '../../../helpers/mountWithTheme';

describe('<FormStructureStyled />', () => {

    it('matches snapshot', () => {
        const wrapper = mountWithTheme(<FormStructureStyled />);
        expect(wrapper.html()).toMatchSnapshot();
    });

});
