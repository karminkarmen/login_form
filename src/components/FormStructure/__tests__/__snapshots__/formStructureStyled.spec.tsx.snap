// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FormStructureStyled /> matches snapshot 1`] = `
"<styled.div>
  <div className=\\"sc-AxjAm dRzAgQ\\" />
</styled.div>"
`;
